src/Contract/ResetPasswordControllerTrait.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Contract;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use SymfonyCasts\Bundle\ResetPassword\ResetPasswordHelperInterface;
+
+trait ResetPasswordControllerTrait
+{
+    private function setCanCheckEmailInSession(Request $request, ResetPasswordHelperInterface $helper, bool $value = true): void
+    {
+        $request->getSession()->set($helper->getSessionEmailKey(), $value);
+    }
+
+    private function isAbleToCheckEmail(SessionInterface $session, ResetPasswordHelperInterface $helper): bool
+    {
+        $sessionKey = $helper->getSessionEmailKey();
+
+        if ($session->get($sessionKey)) {
+            $session->remove($sessionKey);
```

Comment:
~~what if the key doesn't exist?~~ returns null if key doesn't exist.. as would be the case joe blow trys to access the page without a valid request set.